tests/checkpoint/test_checkpoint_with_fluent_datasources.py

```diff
@@ -535,7 +477,6 @@ def test_newstyle_checkpoint_instantiates_and_produces_a_validation_result_when_
         ]
     )
 
-    assert len(context.validations_store.list_keys()) == 4
```

Comment:
Are these `list_keys` asserts testing to make sure that we have stored the appropriate number of results? Even though we aren't using the action, do we still want to assert somehow that the results are being stored?

Reply:
Yup. These are legacy checkpoints so the removal of the action means they aren't persisting anything at all.

The behavior of validation definitions writing directly to stores is unique to V1 checkpoints (v0.18 checkpoints haven't been rewritten to leverage this logic).